lib/facter/framework/core/fact_loaders/fact_loader.rb

```diff
@@ -84,7 +84,7 @@ def construct_reject_lists(blocked_facts, facts)
 
       blocked_facts.each do |blocked|
         facts.each do |fact|
-          next unless fact.name =~ /^#{blocked}/
+          next unless fact.name =~ /^#{blocked}\..*/ || fact.name =~ /^#{blocked}$/
```

Comment:
Instead of doing 2 matches, why not compact this into a single regex?(that would be faster)
```ruby
fact.name =~ /^#{blocked}\..*|^#{blocked}$/ 
```